### PR TITLE
Remove all depercated methods as this is a major release (#1184)

### PR DESCRIPTION
**Pull request by p365labs**
Following the #1184 , I've removed all the deprecated methods inside the project. For the moment I left the method _log inside Client class, as we already discussed. We will fix in a next future as @ruflin explained me. Can you have a look @ruflin and tell me if it looks ok or if I need to fix the commit ? tnx

### CODE REVIEW

**Comment by p365labs** [comment]
@ruflin what kind of error is returning codecov?

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ All notable changes to this project will be documented in this file based on the
 - Remove Elastica/Query/Image.php and test/Elastica/Query/ImageTest.php, no more support for image-plugin.
 - Remove Elastica/Query/Filtered.php and test/Elastica/Query/FilteredTest.php and all uses from code.
 - Remove index.merge.policy.merge_factor, and set/get MergePolicy as it looks deprecated from ES 1.6
+- Remove all deprecated method and classes for the new release.
```
Comment (ruflin) [inline CHANGELOG.md]: Even though the list is going to be quite long, we probably have to be here more specific for people that upgrade and list all the methods that were removed. Please also link to the PR.

File: lib/Elastica/Aggregation/DateHistogram.php [hunk 2]
```diff
@@ -11,34 +9,6 @@
...
-     */
-    public function setPreZone($preZone)
-    {
-        throw new DeprecatedException('Option "pre_zone" is deprecated as of ES 1.5 and will be removed in further Elastica releases. Use "time_zone" instead.');
-    }
-
-    /**
-     * Set post-rounding based on interval.
-     *
-     * @deprecated Option "post_zone" is deprecated as of ES 1.5 and will be removed in further Elastica releases. Use "time_zone" instead.
```
Comment (ruflin) [inline lib/Elastica/Aggregation/DateHistogram.php]: We can directly use these comments for the changelog as I think they are quire good. Just replacing `deprecated` by `removed`.

File: lib/Elastica/Index.php [hunk 3]
```diff
@@ -192,23 +192,6 @@ public function deleteDocuments(array $docs)
...
     /**
-     * Optimizes search index.
-     *
-     * Detailed arguments can be found here in the link
-     *
-     * @param array $args OPTIONAL Additional arguments
-     *
-     * @return array Server response
-     *
-     * @deprecated Replaced by forcemerge
```
Comment (ruflin) [inline lib/Elastica/Index.php]: We cannot remove this yet as it was never really released as deprecated. I suggest to keep this in for the next major release and only remove it afterwards.

Comment (ruflin) [inline lib/Elastica/Index.php]: Which also makes me realise, perhaps we should trigger here a deprecated message?

File: lib/Elastica/Query.php [hunk 4]
```diff
@@ -118,21 +118,6 @@ public function getQuery()
...
 
     /**
-     * Set Filter.
-     *
-     * @param \Elastica\Query\AbstractQuery $filter Filter object
-     *
-     * @return $this
-     *
-     * @link    https://github.com/elasticsearch/elasticsearch/issues/7422
-     * @deprecated Use Elastica\Query::setPostFilter() instead, this method will be removed in further Elastica releases
```
Comment (ruflin) [inline lib/Elastica/Query.php]: Need to check if this one was also just deprecated in one of the last commits.

File: lib/Elastica/Query/Builder.php [hunk 5]
```diff
@@ -1,885 +0,0 @@
-<?php
```
Comment (ruflin) [inline lib/Elastica/Query/Builder.php]: It's hard to see it going but I agree its time to remove it, especially as it was not really maintained anymore.

File: lib/Elastica/Index.php (hunk 3 above)
Comment (p365labs) [inline lib/Elastica/Index.php]: maybe a deprecation message could be better in a next release, considering it was never really released as deprecated, I'll just leave @deprecated. What do u guess ?

Comment (p365labs) [inline lib/Elastica/Index.php]: anyway I'll follow your suggestion to trigger a deprecation

File: lib/Elastica/Query.php (hunk 4 above)
Comment (p365labs) [inline lib/Elastica/Query.php]: setFilter has been deprecated long time ago,
have a look here #1138 
I've also had a look on a bunch of commits even inside this one : 47938554412886f61a3b87d655cb7bf78faee24b
which is from 21/07/2015 so it looks like this is very old

**Comment by ruflin** [comment]
@p365labs You probably have to rebase this PR because of https://github.com/ruflin/Elastica/pull/1197 got merged.

**Pull request merged**